cpu/atxmega/periph/spi.c

```diff
@@ -21,13 +21,49 @@
 
 #include "cpu.h"
 #include "cpu_pm.h"
+#include "macros/math.h"
 #include "mutex.h"
 #include "periph/spi.h"
 #include "pm_layered.h"
 
 #define ENABLE_DEBUG 0
 #include "debug.h"
 
+/**
+ * @brief   Helper function to compute a right shift value corresponding
+ *          to dividers
+ * @{
+ */
+static uint8_t _clk_shift(uint32_t freq)
+{
+    /* Atmega datasheets give the following table:
+     * SPI2X    SPR1    SPR0    SCK Frequency
+     * 0        0       0       Fosc/4
+     * 0        0       1       Fosc/16
+     * 0        1       0       Fosc/64
+     * 0        1       1       Fosc/128
+     * 1        0       0       Fosc/2
+     * 1        0       1       Fosc/8
+     * 1        1       0       Fosc/32
+     * 1        1       1       Fosc/64
+     * We can easily sort it by dividers by inverting the SPI2X bit and
+     * taking it as LSB:
+     * Divider  SPR1    SPR0    ~SPI2X  shift
+     * 2        0       0       0       0
+     * 4        0       0       1       1
+     * 8        0       1       0       2
+     * 16       0       1       1       3
+     * 32       1       0       0       4
+     * 64       1       0       1       5
+     * 64       1       1       0       6
+     * 128      1       1       1       7 */
+    uint8_t shift;
```

Comment:
Please update documentation to reflect xmega reference manual.

SPI2X -> CLK2X
SPR1/0 -> prescaler[1:0]
Fosc -> Clkper

![image](https://github.com/RIOT-OS/RIOT/assets/7027887/94c32346-abe2-4d8f-b951-9c540942c55b)